src/arch/arm/kernel/boot.c

```diff
@@ -29,8 +29,16 @@
 #endif
 
 #ifdef ENABLE_SMP_SUPPORT
-/* sync variable to prevent other nodes from booting
- * until kernel data structures initialized */
+/* The SMP boot synchronization uses a simple mechanism based on a global
+ * variable with the well defined initial value 0. Secondary cores simply keep
+ * spinning until the primary node has initialized all kernel structures and
+ * then set this variable to 1. This mechanism works, because the C rules define
+ * that uninitialized global variables are zero. Technically, such variables are
+ * put either in the BSS segment filled with zeros or the data segment. Both
+ * segments are set up by a previous kernel loader. Any runtime initialization
+ * in the kernel boot code wont work for the SMP synchronization due to a race
+ * condition if all nodes start in an undefined order.
```

Comment:
First part is fine, but the second part is a bit verbose without being clear. What about:

"This mechanism only works if the kernel loader initialised the variable to zero correctly."?

But the above could be said of most kernel functionality. If the real problem is checking if BSS and data sections are initialised correctly, shouldn't we add explicit sanity checks for all known sections, instead of checking one random variable which happened to cause issues in the past? That would also do checking for non-SMP,

Reply:
You have a point there. I created this PR for the specific issue I ran into back then. In the end it's a definition of the ABI the kernel expects. I have a bias for not expecting anything, but this come from our "seL4 in ROM" where there is not really a loader. There is just power up ROM code that should better wipe memory contents for security reasons. But that costs boot time that we had concerns about. So it falls to non-ROM boot code that should do this.

Reply:
@axel-h are you able to reduce this comment and the RISC-V one to just the first part:
```
The SMP boot synchronization uses a simple mechanism based on a global
 * variable with the well defined initial value 0. Secondary cores simply keep
 * spinning until the primary node has initialized all kernel structures and
 * then set this variable to 1.
 ```

Reply:
I've made this a bit shorter. But I think it's work mentioning here that BSS must be zero'd by the loader, otherwise this will not work. For most other BSS variables we have some kind if init option, so we don't depend on a specific init values and this is less critical.